routes/user.js

```diff
@@ -0,0 +1,176 @@
+var express = require("express");
+var passwordHash = require("password-hash");
+var app = express();
+const bcrypt = require("bcrypt");
+var { auth } = require("../config/auth");
+var { user } = require("../models/User");
+var { AccessToken } = require("../models/AccessToken");
+
+app.post("/register", async function (req, res) {
+  const salt = await bcrypt.genSalt();
+  const userPassword = await bcrypt.hash(req.body.password, salt);
+
+  if (req.body.password !== req.body.confirmpassword) {
+    res.status(500).json({
+      error: 1,
+      message: "password not matched",
+      data: null,
+    });
+  } else {
+    const user_post = new user({
+      username: req.body.username,
+      firstname: req.body.firstname,
+      lastname: req.body.lastname,
+      password: userPassword,
+      email: req.body.email,
+    });
+    user_post.save(function (err, row) {
+      if (err) {
+        res.status(500).json({
+          error: 1,
+          message: err.message,
+          data: null,
+        });
+      } 
+      else {
+        res.status(200).json({
+          error: 0,
+          message: "saved successfully",
+          data: null,
+        });
+      }
+    });
+  }
+});
+
+app.post("/login", async function (req, res) {
+  user.findOne({ username: req.body.username }, async function (err, results) {
+    var pass = results.password.toString();
+    var userId = results._id;
+    var input_password = pass.toString();
+    var user_password = req.body.password;
+    var tokenId = userId.toString();
+    if (await bcrypt.compare(user_password, input_password)) {
+      const pwd = passwordHash.generate(req.body.password);
+      AccessToken.findOne({ user_id: userId }, function (err, userDetails) {
+        if (userDetails && userDetails._id) {
```

Comment:
details from the access token table will not be userDetails so provide relevant name to the variable